### DISCUSSION TITLE
Download Threema from Webstore

### DISCUSSION THREAD

**Discussion by ridicolos** [ImranR98/Obtainium#1930] [Q&A]
Hello, I purchased Threema from the Webstore and want do download of directly from https://shop.threema.ch/en/download. 

I need to insert my licence key before the download starts. Is there a way to download Threema from there with obtainium? I know I could use the release provided in F-Droid, but I want the version that uses gplay for notifications.

**Comment by DwainZwerg** [comment]
Not yet. You can create an issue, maybe @ImranR98 will find a way to implement it.

**Comment by viktoriasee** [comment]
The app can be downloaded from anyone. To test, just click the above link in non-android device and the download will just start. The license key has to be entered the first time you launch Threema.

**Reply by ridicolos** [reply]
It's nice that they changed it. But I unfortunately still can't add it to Obtainium.

**Reply by bub0-bub0** [reply]
Download via [Obtainium](https://obtainium.imranr.dev/) directly [from Threema's own repository.](https://releases.threema.ch/fdroid/repo/)
- second tab: “Add app”
– using the path https://releases.threema.ch/fdroid/repo
– select in the pulldown menu below it “Overwrite source”
– select from the list: “F-Droid third-party repo”
– “Automatic selection of the APK with the highest version code” [on]

**Comment by DarkCrypt** [comment]
> I need to insert my licence key before the download starts.

Looks like Threema changed their website so users can download the APK without requiring a license key. As viktoriasee stated above, the license key will be required after downloading and installing the APK.

I've tried different ways to add this app to Obtainium using "HTML" as the override source but I always get the error "Could not get ID from APK".

It's possible to add this app using "Direct APK Link" but I'm not sure if this will continue to retrieve updates.

**Reply by ridicolos** [reply]
Thanks for you replay. Yea that works to some extend. But adding it as a direct apk link only works temporary. Because the link will expire after some time. Also the link seems to depend on a specific version. Could not figure out how to do it in some dynamic way. If you managed to that please let me know.
